### PR TITLE
Fixes #26121 - distinguish between Autocompletes

### PR DESCRIPTION
**Pull request by Ron-Lavi**
In addition to the last Autocomplete refactoring:[ consume the Autocomplete as an independent component](https://github.com/theforeman/foreman/pull/6500),
we need a mechanism to distinguish between the different autocomplete components on a page in the Redux store.
Part of the code that was implemented in [filter's forms Autocomplete](https://github.com/theforeman/foreman/pull/6142)

blocked by: https://github.com/theforeman/foreman/pull/6500
@theforeman/ui-ux could you have a look?

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
What have been done in this PR:
- Add ID into every actions payload.
- In reducers, update the state by the given ID.
- Updated fixtures and tests.
- Update selectors.

File: webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js]: Do we really want to encourage developers to use it? imo `id` should be required string.

File: webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteReducer.js
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteReducer.js:34]: I don't see a reason doing `...state[id]` when you override all the fields later.

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteReducer.js:34]: the `results` is missing here,
and in my opinion it's a good convention to have for future changes

File: webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js]: I mostly agree with you,
though I think there are some cases that the consumer doesn't care for the ID -
they just want the autocomplete to work.
In our case, some autocomplete implementation that I would like to replace are not using an id and defining each one will be difficult, 
that's why I chose to use this mechanism to give more choice.

Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js]: Shouldn't it actually suppose to be easy to define an `id` based on the consumer context?

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js]: Before I'll change it, any other thoughts? 
cc @amirfefer @tbrisker

**Review by amirfefer — DISMISSED** [review]
Overall looks good, thanks @Laviro :+1: 
Tested over #6531 and works well

I agree with @sharvit  about the id, I think it should be required and filled by the the context of the consumer.

**Comment by Ron-Lavi** [comment]
Thank @amirfefer 
@sharvit, done, the ID is required now.

**Comment by amirfefer** [comment]
@Laviro note that there are some  related errors

**Comment by Ron-Lavi** [comment]
fixed that @amirfefer :+1:

**Review by amirfefer — DISMISSED** [review]
Thanks @Laviro works well :+1: 
is it blocked by #6500 ?

**Comment by Ron-Lavi** [comment]
Yes @amirfefer, it is :)

**Comment by ares** [comment]
this needs rebase now, @amirfefer if we need this, would you mind rebasing for @Laviro?

**Review by amirfefer — DISMISSED** [review]
Thanks @tbrisker  for rebasing

**Comment by ares** [comment]
travis tests seems to be failing

I saw warnings like
```
console.error node_modules/jest-prop-type-error/index.js:8
    Warning: [react-bootstrap-typeahead] Passing a falsy `emptyLabel` value to hide the menu when the result set is empty is deprecated. Use `renderMenu` to return `null` instead.
```

otherwise I did testing and didn't find any regression, @amirfefer do you happen to know if the travis issue is related?

**Comment by amirfefer** [comment]
@ares , this travis issue is related to patternfly-react and typeahead 
@Laviro has opened a PR already that fix it #6536

**Review by ares — DISMISSED** [review]

**Comment by ares** [comment]
ready to merge once #6536 is in

**Comment by Ron-Lavi** [comment]
Thanks for the help on this one while I was on maternity leave :)
currently blocked by #6536.

**Comment by Ron-Lavi** [comment]
@ares, just rebased.. ready for review

**Review by ares — APPROVED** [review]
The blocking PR is in now and this was rebased. Given it was green before, taking the risk and merging now. Thanks @Laviro!

**Pull request merged**
